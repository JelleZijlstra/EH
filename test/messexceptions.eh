#!/usr/bin/ehi
# Trying to break the exception mechanism

# Unfortunately, library classes are constant variables
UnknownCommandError = 42

# That is merely funny
UnknownCommandError.toString = func: -> "I don't care"
$print 3

UnknownCommandError.new = 42
$print 3
