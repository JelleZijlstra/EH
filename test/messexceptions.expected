Attempt to write to constant variable: EH runtime error
Uncaught exception of type UnknownCommandError: I don't care
No such object member: operator_colon: EH notice
Uncaught exception of type Null: 
